src/catalog.c

```diff
@@ -277,42 +279,50 @@ catalog_reset(void)
 	catalog.database_id = InvalidOid;
 }
 
-const char *
-catalog_get_cache_proxy_name(CacheType type)
-{
-	return cache_proxy_table_names[type];
-}
-
 Oid
 catalog_get_cache_proxy_id(Catalog *catalog, CacheType type)
 {
-	return catalog->caches[type].inval_proxy_id;
-}
+	if (!catalog_is_valid(catalog))
```

Comment:
why is this branch needed? When do we have an invalid catalog and yet can get past `!extension_is_loaded() || !IsTransactionState()`

Reply:
It's invalid to call `get_namespace_oid()` if not in a transaction state. The catalog can be invalid when not in a transaction state.

Reply:
I don't understand: if !IsTransactionState() the the function will return InvalidOid anyway (line 290). When would we ever hit line 297 and yet have an invalid catalog? I.e. I am asking why this branch is not
```
if (!catalog_is_valid(catalog)) 
    return InvalidOid;
```

Reply:
Are you equating `!catalog_is_valid()` with `!IsTransactionState()`? I guess I am not following what you mean. For instance, if the extension is not loaded, you might have an invalid catalog, so you enter this branch. If you are also NOT in a transaction state, we cannot do anything else but return `InvalidOid`. If we are in a transaction state, we can do an uncached lookup without a catalog.